src/main/scala/org/ergoplatform/ErgoConstants.scala

```diff
@@ -0,0 +1,67 @@
+package org.ergoplatform
+
+import org.ergoplatform.ErgoLikeContext
+import sigmastate.{AtLeast, SBigInt, SPrimType}
+
+case class SizeConstant[T: Numeric](value: T, id: Short, description: String) {
+  def get: T = value
+}
+
+/**
+  * Fundamental constants that are used in sigma's logic and checks
+  */
+object ErgoConstants {
+
+  object MaxInputSize extends SizeConstant[Int](1024 * 1024 * 1, 1,
+    "Input size should not be greater then provided value") {
+  }
+
+  object MaxTreeDepth extends SizeConstant[Int](110, 2,
+    "Max tree depth should not be greater then provided value") {
+  }
+
+  object MaxByteArrayLength extends SizeConstant[Int](10000, 3,
+    "Max bytearray length") {
+  }
+
+  object MaxTokens extends SizeConstant[Byte](4, 6,
```

Comment:
ErgoBox.MaxTokens is not being using in Ergo and actually outdated. There's alternative "val MaxAssetsPerBox = 255" in ErgoTransaction. Please set MaxTokens to the same value, and then we'll use MaxTokens everywhere instead of MaxAssetsPerBox.

Reply:
According to whitepaper, "The maximum number of tokens in a box is set to 4", that's what this constant represent. I suppose it will be wrong to change